apps/glusterfs/volume_entry.go

```diff
@@ -952,3 +968,72 @@ func (v *VolumeEntry) cloneVolumeRequest(db wdb.RODB, clonename string) (*execut
 
 	return vcr, sshhost, nil
 }
+
+type MultiClusterError struct {
+	prefix string
+	errors map[string]error
+}
+
+// NewMultiClusterError returns a MultiClusterError with the given
+// prefix text. Prefix text will be used in the error string if
+// more than one error is captured.
+func NewMultiClusterError(p string) *MultiClusterError {
+	return &MultiClusterError{
+		prefix: p,
+		errors: map[string]error{},
+	}
+}
+
+// Add an error originating with cluster `c` to the captured
+// errors map.
+func (m *MultiClusterError) Add(c string, e error) {
+	m.errors[c] = e
+}
+
+// Return the length of the captured errors map.
+func (m *MultiClusterError) Len() int {
+	return len(m.errors)
+}
+
+// Shorten returns a simplified version of the errors that
+// the MultiClusterError may have captured. It returns nil if
+// no errors were captured. It returns itself if more than one
+// error was captured. It returns the original error if only
+// one error was captured.
+func (m *MultiClusterError) Shorten() error {
+	if len(m.errors) == 0 {
+		return nil
+	}
+	switch len(m.errors) {
+	case 0:
+		return nil
+	case 1:
+		for _, err := range m.errors {
+			return err
+		}
```

Comment:
it is slightly counter-intuitive for me to use a for loop to access the only element

Reply:
Me too. Unfortunately, go does not have any good mechanisms for this. This was the most succinct thing I could come up with. Suggestions welcome, though.